frontend/app/(footer)/dashboard/page.tsx

```diff
@@ -47,19 +61,25 @@ export default function Dashboard() {
 
   useEffect(() => {
     const fetchEvents = async () => {
+      await sleep(500);
       const { event, location } = getQueryParams();
-      setSrcLocation(location);
+      if (event === "UNDEFINED") {
```

Comment:
remove magic string and replace with variable or the undefined keyword in typescript